src/node/snapshotter.h

```diff
@@ -194,8 +178,11 @@ namespace ccf
         msg->data.snapshot = network.tables->snapshot(idx);
 
         last_snapshot_idx = idx;
+
+        static uint32_t generation_count = 0;
         threading::ThreadMessaging::thread_messaging.add_task(
-          get_execution_thread(), std::move(msg));
+          threading::ThreadMessaging::get_execution_thread(generation_count++),
+          std::move(msg));
```

Comment:
you probably do not want to run on thread 0 if there is more than 1 thread

Reply:
Doesn't `++tid;` in `thread_messaging::get_execution_thread()` prevent that?

Reply:
yes you are right